scripts/test/nightly/updateAndRunTests.sh

```diff
@@ -19,12 +19,48 @@ set -e
 unset CDPATH
 
 THIS_DIR=$(cd "$(dirname "$0")" && pwd)
+REPO_DIR=$(cd "$THIS_DIR" && cd ../../.. && pwd)
+
+if [ "$#" -ne 2 ]; then
+  echo "Usage: $0 <settings-json-file> <test-type>"
+  echo "Valid test types are: performance, functional"
+  exit 1
+fi
+
+SETTINGS_FILE=$1
+TEST_TYPE=$2
+
+VPC=$(jq ".vpc" "$SETTINGS_FILE" --raw-output)
+SUBNETS=$(jq ".subnets" "$SETTINGS_FILE" --raw-output)
+RESULTS_BUCKET=$(jq ".resultsBucket" "$SETTINGS_FILE" --raw-output)
+REPO_PATH=$(jq ".repoPath" "$SETTINGS_FILE" --raw-output)
+MERGE_TO_MAIN=$(jq ".mergeToMainOnTestType.$TEST_TYPE" "$SETTINGS_FILE" --raw-output)
 
 pushd "$THIS_DIR"
 
+git remote set-url origin "https://github.com/$REPO_PATH.git"
 git fetch
 git switch --discard-changes -C develop origin/develop
 
-popd
+set +e
+./runTests.sh "$VPC" "$SUBNETS" "$RESULTS_BUCKET" "$TEST_TYPE"
+EXIT_CODE=$?
+set -e
+
+if [ $EXIT_CODE -eq 0 ] && [ "$MERGE_TO_MAIN" == "true" ]; then
+  echo "Will merge changes into main branch..."
+  PRIVATE_KEY_FILE=$(jq ".gitHubApp.privateKeyFile" "$SETTINGS_FILE" --raw-output)
+  APP_ID=$(jq ".gitHubApp.appId" "$SETTINGS_FILE" --raw-output)
+  INSTALLATION_ID=$(jq ".gitHubApp.installationId" "$SETTINGS_FILE" --raw-output)
+  # Copy merge to main script to keep the version from develop
+  TMP_MERGE_TO_MAIN=$(mktemp)
+  cat ./mergeToMain.sh >> "$TMP_MERGE_TO_MAIN"
+  chmod u+x "$TMP_MERGE_TO_MAIN"
+  pushd "$REPO_DIR"
+  "$TMP_MERGE_TO_MAIN" "$REPO_PATH" "$PRIVATE_KEY_FILE" "$APP_ID" "$INSTALLATION_ID"
+  popd
+else
+  echo "Not merging changes into main branch"
```

Comment:
We could log out here why we're not merging into main? Whether that be the exit code was not 0 or we're running a test type that is configured to not merge into main

Reply:
Good idea, done